main.go

```diff
@@ -1,143 +1,92 @@
-	serv.socketHub.WithConnectionCallback(serv.newClientConnection)
-
-	// Start websocket, router and Thea
-	wg := &sync.WaitGroup{}
-	wg.Add(3)
-	go func() {
-		defer wg.Done()
-		serv.socketHub.Start()
-	}()
-	go func() {
-		defer wg.Done()
-		serv.httpRouter.Start(&api.RouterOptions{
-			ApiPort: 8080,
-			ApiHost: "0.0.0.0",
-			ApiRoot: "/api/thea",
-		})
-	}()
-	go func() {
-		defer wg.Done()
-		if err := serv.thea.Start(); err != nil {
-			mainLogger.Emit(logger.FATAL, "Failed to start Processor: %v", err.Error())
-		}
-
-		mainLogger.Emit(logger.STOP, "Processor shutdown, cleaning up supporting services...\n")
-		serv.socketHub.Close()
-		serv.httpRouter.Stop()
-	}()
+func startThea(config *internal.TheaConfig) {
+	log.Emit(logger.INFO, " --- Starting Thea (version %.1f) ---\n", VERSION)
 
-	// Wait for all processes to finish
-	wg.Wait()
-}
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	go listenForInterrupt(ctxCancel)
 
-// setupRoutes initialises the routes and commands for the HTTP
-// REST router, and the websocket hub
-func (serv *services) setupRoutes() {
-	serv.httpRouter.CreateRoute("v0/queue", "GET", serv.httpGateway.HttpQueueIndex)
-	serv.httpRouter.CreateRoute("v0/queue/{id}", "GET", serv.httpGateway.HttpQueueGet)
-	serv.httpRouter.CreateRoute("v0/queue/promote/{id}", "POST", serv.httpGateway.HttpQueueUpdate)
-	serv.httpRouter.CreateRoute("v0/ws", "GET", serv.socketHub.UpgradeToSocket)
-
-	serv.socketHub.BindCommand("QUEUE_INDEX", serv.wsGateway.WsQueueIndex)
-	serv.socketHub.BindCommand("QUEUE_DETAILS", serv.wsGateway.WsQueueDetails)
-	serv.socketHub.BindCommand("QUEUE_REORDER", serv.wsGateway.WsQueueReorder)
-	serv.socketHub.BindCommand("TROUBLE_RESOLVE", serv.wsGateway.WsTroubleResolve)
-	serv.socketHub.BindCommand("TROUBLE_DETAILS", serv.wsGateway.WsTroubleDetails)
-	serv.socketHub.BindCommand("PROMOTE_ITEM", serv.wsGateway.WsItemPromote)
-	serv.socketHub.BindCommand("PAUSE_ITEM", serv.wsGateway.WsItemPause)
-	serv.socketHub.BindCommand("CANCEL_ITEM", serv.wsGateway.WsItemCancel)
-
-	serv.socketHub.BindCommand("PROFILE_INDEX", serv.wsGateway.WsProfileIndex)
-	serv.socketHub.BindCommand("PROFILE_CREATE", serv.wsGateway.WsProfileCreate)
-	serv.socketHub.BindCommand("PROFILE_REMOVE", serv.wsGateway.WsProfileRemove)
-	serv.socketHub.BindCommand("PROFILE_MOVE", serv.wsGateway.WsProfileMove)
-	serv.socketHub.BindCommand("PROFILE_SET_MATCH_CONDITIONS", serv.wsGateway.WsProfileSetMatchConditions)
-	serv.socketHub.BindCommand("PROFILE_UPDATE_COMMAND", serv.wsGateway.WsProfileUpdateCommand)
-}
+	if err := internal.New(*config).Run(ctx); err != nil {
+		log.Emit(logger.FATAL, "Failed to start Thea: %v", err.Error())
 
-func (serv *services) handleTheaUpdate(update *internal.Update) {
-	body := map[string]interface{}{"context": update}
-	if update.UpdateType == internal.PROFILE_UPDATE {
-		body["profiles"] = serv.thea.GetAllProfiles()
-		body["targetOpts"] = serv.thea.GetKnownFfmpegOptions()
+		return
```

Comment:
```suggestion
		log.Emit(logger.FATAL, "Thea failed unexpectedly: %v", err)
        os.Exit(1)
```

This block runs at the end of the program, so a different error message might make more sense.

Reply:
And actually `startThea` should probably be `runThea`

Reply:
Any reason to prefer run over start?